File: desafio-02/LeonardRuhmann/javascript/primos.js
const primos = [];

for (let numero = 2; numero <= 1000; numero++) {
    let ehPrimo = true;
    const raiz = Math.sqrt(numero);
    for (let i = 0; i < primos.length && primos[i] <= raiz; i++) {
        if (numero % primos[i] === 0) {
            ehPrimo = false;
            break;
        }
    }
    if (ehPrimo) {
        primos.push(numero);
        console.log(numero);
    }
};

File: desafio-02/flafmg/lua/primos.lua
function eh_primo(n)
    if n < 2 then return false end
    for i = 2, math.sqrt(n) do
        if n % i == 0 then
            return false
        end
    end
    return true
end

for i = 1, 10000 do
    if eh_primo(i) then
        print(i)
    end
end